engine/job.go

```diff
@@ -21,32 +22,34 @@ type Job interface {
 	Delay() uint32
 	Tries() uint16
 	ElapsedMS() int64
+	Attributes() map[string]string
 
 	encoding.BinaryMarshaler
 	encoding.BinaryUnmarshaler
 	encoding.TextMarshaler
 }
 
 type jobImpl struct {
-	namespace string
-	queue     string
-	id        string
-	body      []byte
-	ttl       uint32
-	delay     uint32
-	tries     uint16
+	namespace  string
+	queue      string
+	id         string
+	body       []byte
+	ttl        uint32
+	delay      uint32
+	tries      uint16
+	attributes map[string]string
 
 	_elapsedMS int64
 }
 
 // NOTE: there is a trick in this factory, the delay is embedded in the jobID.
 // By doing this we can delete the job that's located in hourly AOF, by placing
 // a tombstone record in that AOF.
-func NewJob(namespace, queue string, body []byte, ttl, delay uint32, tries uint16, jobID string) Job {
+func NewJob(namespace, queue string, body []byte, ttl, delay uint32, tries uint16, jobID string, attr string) Job {
```

Comment:
I think we can extend a function like `AddAttribute` for the job.

Reply:
Because the job data marshaling is in NewJob function, I think it's quite nature to put body and attributes together since these two are the input of marshaling.
If we leave attributes somewhere outside NewJob, then we will need to extract marshaling out of NewJob.

Reply:
What I mean is NewJob parameter is too long now and we can add a new method like `AddAttribute` for the job object NOT outside.